modules/cassandrabackend/cassandrabackend.cc

```diff
@@ -0,0 +1,324 @@
+/*
+ * CassandraBackend - a high performance Cassandra based backend for PowerDNS written by
+ * Sumit Kumar, 2015
+ *
+ * The license will be the same as the ones used in powerdns authorative server
+ */
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+#include "pdns/utility.hh"
+#include "pdns/dnsbackend.hh"
+#include "pdns/dns.hh"
+#include "pdns/dnspacket.hh"
+#include "pdns/pdnsexception.hh"
+#include "pdns/logger.hh"
+#include <signal.h>
+#include "pdns/arguments.hh"
+#include "pdns/base32.hh"
+#include "pdns/lock.hh"
+#include <iostream>
+#include "cassandradbmanager.h"
+#include <string.h>
+#include <boost/tokenizer.hpp>
+#include <boost/foreach.hpp>
+
+#if 0
+#define DEBUGLOG(msg) L<<Logger::Error<<msg
+#else
+#define DEBUGLOG(msg) do {} while(0)
+#endif
+
+class CassandraBackend : public DNSBackend
+{
+
+private:
+  backendrecord* backendRecords = NULL;
+  int recordIndex = 0;
+  int totalSize = 0;
+  domainlookuprecords record;
+  std::string domain;
+  std::string queryType;
+  int all_authoritative_server = 0;
+
+  bool hasEnding (std::string const &fullString, std::string const &ending) {
+      if (fullString.length() >= ending.length()) {
+          return (0 == fullString.compare (fullString.length() - ending.length(), ending.length(), ending));
+      } else {
+          return false;
+      }
+  }
+public:
+  CassandraBackend(const string &suffix) {
+	  L << Logger::Info << " Launching CassandraBackend instance " << endl;
+	  setArgPrefix("cassandra"+suffix);
+	 try {
+		std::string seed_nodes = getArg("seed-nodes");
+		std::string username = getArg("username");
+		std::string password = getArg("password");
+		std::string keyspace = getArg("keyspace");
+		int core_connections = getArgAsNum("core-connections");
+		int max_connections = getArgAsNum("max-connections");
+		int max_concurrent_creations = getArgAsNum("max-concurrent-creations");
+		int num_io_threads = getArgAsNum("num-io-threads");
+		int protocol_version = getArgAsNum("protocol-version");
+		int queue_size_io = getArgAsNum("queue-size-io");
+		int queue_size_event = getArgAsNum("queue-size-event");
+		int reconnect_wait_time = getArgAsNum("reconnect-wait-time");
+		int concurrent_requests_threshold = getArgAsNum("concurrent-requests-threshold");
+		int connect_timeout = getArgAsNum("connect-timeout");
+		int request_timeout = getArgAsNum("request-timeout");
+		int enable_load_balance_round_robin = getArgAsNum("enable-load-balance-round-robin");
+		int enable_token_aware_routing = getArgAsNum("enable-token-aware-routing");
+		int enable_latency_aware_routing = getArgAsNum("enable-latency-aware-routing");
+		int enable_tcp_nodelay = getArgAsNum("enable-tcp-nodelay");
+		int enable_tcp_keepalive = getArgAsNum("enable-tcp-keepalive");
+		all_authoritative_server = getArgAsNum("all-authoritative-server");
```

Comment:
Can you please explain this feature and why it is required? It seems bit odd place for it.